### PR TITLE
feat(list-item): add `unavailable` property

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #5664

## Summary

- add `unavailable` property to `list-item`.
- add e2e test
- add entry in story
- update demo

### CODE REVIEW

**Review by ashetland — APPROVED** [review]
Visuals lookin' good. Can we add an example of `unavailable` to the start-and-end-content-slots text?

**Comment by driskull** [comment]
@ashetland updated stories.

File: packages/calcite-components/src/components/list-item/list-item.tsx [hunk 1]
```diff
@@ -196,6 +196,11 @@ export class ListItem
     this.calciteInternalListItemSelect.emit();
   }
 
+  /**
+   * When `true`, the component's content appears inactive.
```
Comment (jcfranco) [inline packages/calcite-components/src/components/list-item/list-item.tsx:200]: +@geospatialem @DitwanP for doc review

**Review by jcfranco — APPROVED** [review]
✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️
✖️✖️⛔⛔⛔✖️✖️⛔⛔✖️✖️✖️⛔⛔✖️✖️⛔✖️✖️✖️⛔✖️
✖️⛔✖️✖️✖️✖️⛔✖️✖️⛔✖️⛔✖️✖️⛔✖️⛔✖️✖️✖️⛔✖️
✖️⛔✖️✖️✖️✖️⛔✖️✖️⛔✖️⛔✖️✖️⛔✖️⛔✖️✖️✖️⛔✖️
✖️⛔✖️✖️✖️✖️⛔✖️✖️⛔✖️⛔✖️✖️⛔✖️⛔✖️✖️✖️✖️✖️
✖️✖️⛔⛔⛔✖️✖️⛔⛔✖️✖️✖️⛔⛔✖️✖️⛔⛔⛔✖️⛔✖️
✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️✖️

File: packages/calcite-components/src/components/list-item/list-item.tsx (hunk 1 above)
Comment (driskull) [reply packages/calcite-components/src/components/list-item/list-item.tsx:200]: @geospatialem @DitwanP i'll merge now. We can update the doc post install if we need to.

Comment (geospatialem) [reply packages/calcite-components/src/components/list-item/list-item.tsx:200]: The doc LGTM! We can add a new sample to showcase the behavior for folks once `2.13.0` is live for additional context as well. ⚡

**Pull request merged**